chat/models.py

```diff
@@ -66,38 +66,95 @@ def __str__(self):
     def get_service(self):
         return settings.CHAT_SERVICES[self.service].get_instance()
 
+    def get_guild_id(self):
+        return self.puzzle.hunt.settings.discord_guild_id
+
+    def _get_category_name(self):
+        if self.puzzle.is_meta:
+            return self.puzzle.hunt.settings.discord_metas_category
+
+        metas = self.puzzle.metas.order_by("created_on").all()
+        if len(metas) > 0:
+            return metas[0].name  # default to the oldest created meta
```

Comment:
This seems fine, but one suggestion to consider: for puzzles assigned to multiple metas, put it in the oldest _unsolved_ meta, and when that meta gets solved, move the puzzle to the oldest remaining unsolved meta